src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1569,16 +1569,14 @@ public function getPreferredFormat(?string $default = 'html'): ?string
             return $this->preferredFormat;
         }
 
-        $this->preferredFormat = $this->getRequestFormat($this->getContentType());
-
-        if (null === $this->preferredFormat) {
-            foreach ($this->getAcceptableContentTypes() as $contentType) {
-                if (null !== $this->preferredFormat = $this->getFormat($contentType)) {
-                    break;
-                }
+        foreach ($this->getAcceptableContentTypes() as $contentType) {
+            if ($this->preferredFormat = $this->getFormat($contentType)) {
+                return $this->preferredFormat;
             }
         }
 
+        $this->preferredFormat = $this->getRequestFormat($this->getContentType());
```

Comment:
The previous order was: `_format` attribute, `Content-Type` and `Accept` headers,
Now it is: `Accept`, `_format` attribute, `Content-Type`,

I think we should keep the `_format` as the first alternative, otherwise it will never be reached in normal requests through the browser (where `Accept` is always sent).

Reply:
Good catch I’ll update